ext/uri/php_uri.c

```diff
@@ -136,7 +136,11 @@ ZEND_ATTRIBUTE_NONNULL static zend_result php_uri_get_property(const uri_interna
 		return FAILURE;
 	}
 
-	return property_handler->read_func(internal_uri, read_mode, zv);
+	zend_result result = property_handler->read_func(internal_uri, read_mode, zv);
+
+	ZEND_ASSERT(result == FAILURE || (Z_TYPE_P(zv) == IS_STRING && GC_REFCOUNT(Z_STR_P(zv)) == 2) || Z_TYPE_P(zv) == IS_NULL || Z_TYPE_P(zv) == IS_LONG);
```

Comment:
this assertions intends to make sure that only a new string, null, or long can be returned in case of success. I hope that you won't find it too obtrusive.